### ISSUE TITLE
Can not get svc after updated to latest code

### ISSUE THREAD

**Issue by hurf** [GoogleCloudPlatform/kubernetes#9462]
I created a pod and a service for testing purpose with v1beta1 template. And they all worked fine. The k8s binaries are compiled from code 2 weeks ago.

    $ kubectl get po -l type=apache2
    POD       IP           CONTAINER(S)   IMAGE(S)        HOST                  LABELS                        STATUS    CREATED      MESSAGE
    apache1   172.17.0.1                                  127.0.0.1/127.0.0.1   tier=webserver,type=apache2   Running   4 minutes    
                           apache1        test/apache:1                                                       Running   4 minutes  
    $ kubectl get svc -l name=webserver
    NAME        LABELS           SELECTOR                      IP(S)        PORT(S)
    webserver   name=webserver   tier=webserver,type=apache2   11.1.1.169   999/TCP

Then I pulled the latest code of k8s. But `kubectl get svc` doesn't work. 

    $ kubectl get po
    NAME      READY     REASON    RESTARTS   AGE
    apache1   1/1       Running   1          2h
    $ kubectl get svc
    Error from server: no kind "Service" is registered for version "v1beta1"
    
Then I tried to create a service with v1beta3 template, the creation seems completed, but  `get svc` still not working.

    $ kubectl create -f servicev3.yaml 
    services/v3svc
    $ kubectl get svc
    Error from server: no kind "Service" is registered for version "v1beta1"

Any idea on this problem?

**Comment by hurf** [comment]
I used `kubectl describe svc`, found both services exsits. But no endpoint is attached to the newly created one.

new:

    Name:			v3svc
    Labels:			name=webserver
    Selector:		tier=webserver,type=apache2
    Type:			ClusterIP
    IP:			11.1.1.158
    Port:			<unnamed>	999/TCP
    Endpoints:		<none>
    Session Affinity:	None
    No events.

old:

    Name:			webserver
    Labels:			name=webserver
    Selector:		tier=webserver,type=apache2
    Type:			ClusterIP
    IP:			11.1.1.169
    Port:			<unnamed>	999/TCP
    Endpoints:		172.17.0.1:80
    Session Affinity:	None
    No events.

Then I deleted both services, and recreated the new one, still no endpoint is attached. Also `get svc` is not working.
I tried to find the IP address of pod apache1, but neither `get po` nor `describe po` contains the IP address of the pod now.

**Comment by hurf** [comment]
Find the cause, the registry is not updated in etcd. After deleted /registry directory manually, k8s generated new keys automatically, all works fine now. The lost is all data are gone, never do it in a productive environment.

    $ etcdctl get /registry/namespaces/default
    {"kind":"Namespace","id":"default","uid":"8b05a41a-c633-11e4-be1c-fa163e837eb3","creationTimestamp":"2015-03-09T08:09:00Z","apiVersion":"v1beta1","spec":{},"status":{}}
    $ etcdctl rm --recursive /registry
    $ etcdctl get /registry/namespaces/default
    {"kind":"Namespace","apiVersion":"v1","metadata":{"name":"default","uid":"530e001a-1329-11e5-a04e-fa163e837eb3","creationTimestamp":"2015-06-15T06:39:51Z"},"spec":{"finalizers":["kubernetes"]},"status":{"phase":"Active"}}

**Comment by cameronbrunner** [comment]
I know this is closed now but I just ran into this problem as well when upgrading from v1beta3 to v1.  I was able to track it down to two specific entries in the registry, `/registry/services/endpoints/default/kubernetes-ro` and `/registry/services/specs/default/kubernetes-ro`.  Deleting these with `kubectl` was not possible but I was able to remove them with `etcdctl`.  Once both entries were removed services began working properly again.
